test/e2e/apps/statefulset.go

```diff
@@ -2178,8 +2180,16 @@ func verifyStatefulSetPVCsExistWithOwnerRefs(ctx context.Context, c clientset.In
 }
 
 func expectPodNames(pods *v1.PodList, expectedPodNames []string) {
-	framework.ExpectEqual(len(pods.Items), len(expectedPodNames), "unexpected number of pods")
-	for i, pod := range pods.Items {
-		framework.ExpectEqual(pod.Name, expectedPodNames[i], "unexpected pod name")
+	actualPods := []string{}
+	for _, pod := range pods.Items {
+		// ignore terminating pods, similarly to how the controller does it
+		// when calculating status information
+		if e2epod.IsPodActive(&pod) {
+			actualPods = append(actualPods, pod.Name)
+		}
+	}
+	if !reflect.DeepEqual(expectedPodNames, actualPods) {
```

Comment:
do you have guarantee the order or should you sort both slices first?

Reply:
Usually those are sorted before invoking `expectPodNames`, it's not plainly expressed in the doc, I could probably move the sorting into this function instead to ensure that the list gets sorted and explain that the expected list should be sorted, if you think that makes the code easier.

Reply:
as you wish, but at least add a comment explaining that `actualPods` and `expectedPodNames` are expected to have the same order